compiler/src/dmd/toir.d

```diff
@@ -882,11 +890,23 @@ void buildClosure(FuncDeclaration fd, IRState *irs)
         if (driverParams.symdebug)
             toDebugClosure(Closstru.Ttag);
 
+        // Add extra size so we can align it
+        if (aggAlignment > 16)  // gc aligns on 16 bytes
+            structsize += aggAlignment - 16;
+
         // Allocate memory for the closure
         elem *e = el_long(TYsize_t, structsize);
         e = el_bin(OPcall, TYnptr, el_var(getRtlsym(RTLSYM.ALLOCMEMORY)), e);
         toTraceGC(irs, e, fd.loc);
 
+        // Align it
+        if (aggAlignment > 16)
```

Comment:
The magic number `16` could be replaced by an enum constant

Reply:
I will in the followup.